src/openapi/transport/batch.ts

```diff
@@ -0,0 +1,292 @@
+            (typeof batchResponse.isNetworkError === 'boolean'
+                ? batchResponse.isNetworkError
+                : !batchResponse.status);
+
+        const logFunction =
+            isAuthFailure || isNetworkError ? log.debug : log.error;
+        logFunction(LOG_AREA, 'Batch request failed', batchResponse);
+
+        for (let i = 0; i < callList.length; i++) {
+            // pass on the batch response so that if a batch responds with a 401,
+            // and queue is before batch, queue will auto retry
+            callList[i].reject({
+                message: 'batch failed',
+                status: isAuthFailure ? 401 : undefined,
+                isNetworkError,
+            });
+        }
+    };
+
+    private batchCallSuccess = (
+        callList: QueueItem[],
+        batchResult: BatchResult,
+    ) => {
+        // Previously occurred due to a bug in the auth transport
+        if (!(batchResult && batchResult.response)) {
+            log.error('Received success call without response', batchResult);
+            this.batchCallFailure(callList, batchResult);
+            return;
+        }
+
+        const parentRequestId = getParentRequestId(batchResult);
+
+        const results = parseBatch(batchResult.response, parentRequestId);
+
+        for (let i = 0; i < callList.length; i++) {
+            const call = callList[i];
+            const result = results[i];
+            if (result) {
+                // decide in the same way as transport whether the call succeeded
+                if (
+                    result.status &&
+                    (result.status < 200 || result.status > 299) &&
+                    result.status !== 304
+                ) {
+                    call.reject(result);
+                } else {
+                    call.resolve(result);
+                }
+            } else {
+                log.error(LOG_AREA, 'A batch response was missing', {
+                    index: i,
+                    ...batchResult,
+                });
+                call.reject();
+            }
+        }
+    };
+
+    private runBatches() {
+        this.nextTickTimer = false;
+        const serviceGroupMap = this.emptyQueueIntoServiceGroups();
+        const serviceGroups = Object.keys(serviceGroupMap);
+        for (let i = 0, l = serviceGroups.length; i < l; i++) {
+            const serviceGroupList = serviceGroupMap[serviceGroups[i]];
+            if (serviceGroupList.length === 1) {
+                this.runQueueItem(serviceGroupList[0]);
+            } else {
+                this.runBatchCall(serviceGroups[i], serviceGroupList);
+            }
+        }
+    }
+
+    addToQueue(item: QueueItem) {
+        super.addToQueue(item);
+        if (!this.nextTickTimer || this.timeoutMs > 0) {
+            if (this.timeoutMs === 0) {
+                this.nextTickTimer = true;
+                // @ts-ignore

```

Comment:
please update the type of the nextTick instead

Reply:
whats about replacing this with setTimeout(callback,0)?

Reply:
this is one of the option in the nextTick helper